### PR TITLE
Decompile overlay_25

### PR DESCRIPTION
**Pull request by cbt6**

### CODE REVIEW

File: src/overlay_25.c [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+        DestroyMsgData(messageData);
+        GF_ASSERT(otName);
+        if (otName) {
+            u32 length = String_GetLength(otName);
+            GF_ASSERT(length <= PLAYER_NAME_LENGTH);
+            if (length > PLAYER_NAME_LENGTH) {
+                length = PLAYER_NAME_LENGTH;
+            }
+            MIi_CpuClear16(0xffff, set->trainer.otName, (PLAYER_NAME_LENGTH + 1) * sizeof(u16));
+            MIi_CpuCopy16(String_cstr(otName), set->trainer.otName, length * sizeof(u16));
```
Comment (tgsm) [inline src/overlay_25.c:175]:
```suggestion
            MI_CpuFill16(set->trainer.otName, 0xffff, (PLAYER_NAME_LENGTH + 1) * sizeof(u16));
            MI_CpuCopy16(String_cstr(otName), set->trainer.otName, length * sizeof(u16));
```

File: src/overlay_25.c [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        String *name = GetSpeciesName(trainerHouseMon->species, HEAP_ID_FIELD);
+        GF_ASSERT(name);
+        if (name) {
+            u32 length = String_GetLength(name);
+            GF_ASSERT(length <= POKEMON_NAME_LENGTH);
+            if (length > POKEMON_NAME_LENGTH) {
+                length = POKEMON_NAME_LENGTH;
+            }
+            MIi_CpuClear16(0xffff, trainerHouseMon->nickname, POKEMON_NAME_LENGTH * sizeof(u16));
+            MIi_CpuCopy16(String_cstr(name), trainerHouseMon->nickname, length * sizeof(u16));
```
Comment (tgsm) [inline src/overlay_25.c:193]:
```suggestion
            MI_CpuFill16(trainerHouseMon->nickname, 0xffff, POKEMON_NAME_LENGTH * sizeof(u16));
            MI_CpuCopy16(String_cstr(name), trainerHouseMon->nickname, length * sizeof(u16));
```

File: src/overlay_25.c [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+BOOL ScrCmd_ShowTrainerHouseIntroMessage(ScriptContext *ctx) {
+    u16 trainerNum = ScriptGetVar(ctx);
+    TrainerHouse *trainerHouse = Save_TrainerHouse_Get(ctx->fieldSystem->saveData);
+    MAIL_MESSAGE intro;
+    MAIL_MESSAGE temp;
+    if (trainerNum == MAX_NUM_TRAINER_HOUSE_SETS) {
+        MailMsg_Init_FromTemplate(&temp, &TrainerHouse_DefaultIntroMessage);
+        MIi_CpuCopy16((u16*)&temp, (u16*)&intro, sizeof(MAIL_MESSAGE));
+    } else {
+        MIi_CpuCopy16((u16*)&(trainerHouse->sets[trainerNum].trainer.introMessage), (u16*)&intro, sizeof(MAIL_MESSAGE));
```
Comment (tgsm) [inline src/overlay_25.c:208]:
```suggestion
        MI_CpuCopy16(&temp, &intro, sizeof(MAIL_MESSAGE));
    } else {
        MI_CpuCopy16(&(trainerHouse->sets[trainerNum].trainer.introMessage), &intro, sizeof(MAIL_MESSAGE));
```

**Review by tgsm — CHANGES_REQUESTED** [review]

**Review by tgsm — DISMISSED** [review]

**Review by red031000 — CHANGES_REQUESTED** [review]
few small things, need discussing whether appropriate

File: src/overlay_25.c [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    return TRUE;
+}
+
+static BattleSetup *TrainerHouse_NewBattleSetup(FieldSystem *fieldSystem, TrainerHouseSet *set) {
+    s32 i;
+    BattleSetup *setup = BattleSetup_New(HEAP_ID_FIELD, BATTLE_TYPE_TRAINER | BATTLE_TYPE_TOWER | BATTLE_TYPE_13);
+    SaveData *saveData = fieldSystem->saveData;
+    Party *party = SaveArray_Party_Get(saveData);
+    sub_02051D18(setup, fieldSystem, saveData, fieldSystem->location->mapId, fieldSystem->unk94, fieldSystem->unkB0);
+    setup->battleBg = 6;
```
[the remark this answers is not in the document]
Comment (cbt6) [reply src/overlay_25.c:233]: To be done in a separate PR.

**Review by red031000 — CHANGES_REQUESTED** [review]
I dont see where you renamed the scrcmd in scrcmd.c? I see it in the headers but not in the c file? possible that github is playing up?

**Comment by cbt6** [comment]
Are you referring to `ScrCmd_ShowTrainerHouseIntroMessage`? That's in `src/overlay_25.c`

**Comment by red031000** [comment]
odd that it's not in an scrcmd file

**Review by red031000 — APPROVED** [review]

**Pull request merged**